patches/tModLoader/Terraria/ModLoader/TileLoader.cs

```diff
@@ -686,6 +691,37 @@ public static void ModifyLight(int i, int j, int type, ref float r, ref float g,
 		}
 	}
 
+	/// <summary>
+	/// Registers a tile type as having custom biome conversion code for this specific <see cref="BiomeConversionID"/>. For modded tiles, you can directly use <see cref="Convert"/>
+	/// </summary>
+	/// <param name="tileType">The tile type that has is affected by this custom conversion.</param>
+	/// <param name="conversionType">The conversion type for which the tile should use custom conversion code.</param>
+	/// <param name="conversionDelegate">Code to run when the tile attempts to get converted. Return false to signal that your custom conversion took place and that vanilla code shouldn't be ran.</param>
+	public static void RegisterConversion(int tileType, int conversionType, ConvertTile conversionDelegate)
+	{
+		var conversions = tileConversionDelegates[tileType] ??= new List<ConvertTile>[BiomeConversionLoader.BiomeConversionCount];
```

Comment:
Perhaps we should make sure that `RegisterConversion` can only be called after `ResizeArrays`, otherwise `BiomeConversionCount` won't be set in stone, and a mod calling `RegisterConversion` later for the same tile may throw (and not be at fault)

The easiest way is probably just to set `tileConversionDelegates = null` by default and in `Unload` and fully reallocate it in `ResizeArrays`